boot/boot_serial/src/boot_serial.c

```diff
@@ -439,6 +439,45 @@ bs_upload(char *buf, int len)
     flash_area_close(fap);
 }
 
+#ifdef MCUBOOT_BOOT_MGMT_ECHO
+static bool 
+decode_echo(cbor_state_t *state, cbor_string_type_t *result)
+{
+    size_t bsstrdecoded;
+
+    if (!map_start_decode(state)) {
+        return false;
+    }
+    if (!multi_decode(2, 2, &bsstrdecoded, (void *)tstrx_decode, state, result, sizeof(cbor_string_type_t))) {
+        return false;
+    }
+    map_end_decode(state);
+    return true;
+}
+
+
+static void
+bs_echo(char *buf, int len)
+{
+    const char *echo_buf="";
+    size_t echo_buflen = strlen(echo_buf);
+    size_t bsstrdecoded;
+
+    cbor_string_type_t str[2];
+
+    if (entry_function((const uint8_t *)buf, len, (const void *)str, &bsstrdecoded, (void *)decode_echo, 1, 2)) {
+        echo_buf = (char*)str[1].value;
+        echo_buflen = str[1].len;
+    }
+
+    map_start_encode(&cbor_state, 10);
+    tstrx_put(&cbor_state, "r");
+    tstrx_encode(&cbor_state, &(cbor_string_type_t){.value = (const uint8_t *)echo_buf, .len = echo_buflen});
```

Comment:
Anonymous definition of `cbor_string_type_t` structure as function parameter makes the code hard to read and nothing more. Don't you actually want to return what has been decoded into `str`? Why not use the `str` directly here?

Reply:
good catch, I've updated the code to just use the found str in the buffer, which skips this anonymous struct afterall.